47.md

```diff
@@ -0,0 +1,147 @@
+    }
+}
+```
+
+The data field SHOULD contain a `message` field with a human readable error message and a `code` field with the error code if the status is `error`.
+
+### Error codes
+- `RATE_LIMITED`: The client is sending commands too fast. It should retry in a few seconds.
+- `NOT_IMPLEMENTED`: The command is not known or is intentionally not implemented.
+- `INSUFFICIENT_BALANCE`: The wallet does not have enough funds to cover a fee reserve or the payment amount.
+- `QUOTA_EXCEEDED`: The wallet has exceeded
+- `RESTRICTED`: This public key is not allowed to do this operation.
+- `UNAUTHORIZED`: This public key has no wallet connected.
+- `INTERNAL`: An internal error.
+- `OTHER`: Other error.
+
+## Nostr Wallet Connect URI
+**client** discovers **wallet service** by scanning a QR code, handling a deeplink or pasting in a URI.
+
+The **wallet service** generates this connection URI with protocol `nostr+walletconnect:` and base path it's hex-encoded `pubkey` with the following query string parameters: 
+
+- `relay` Required. URL of the relay where the **wallet service** is connected and will be listening for events. May be more than one.
+- `secret` Required. 32-byte randomly generated hex encoded string. The **client** should use this to sign events when communicating with the **wallet service**.
+    - Authorization does not require passing keys back and forth.
+    - The user can have different keys for different applications. Keys can be revoked and created at will and have arbitrary constraints (eg. budgets).
+    - The key is harder to leak since it is not shown to the user and backed up.
+    - It improves privacy because the user's main key would not be linked to their payments.
+
+The **client** should then store this connection and use it when the user wants to perform actions like paying an invoice. Due to this NIP using ephemeral events, it is recommended to pick relays that do not close connections on inactivity to not drop events.
+
+### Example connection string
+```sh
+nostr+walletconnect:b889ff5b1513b641e2a139f661a661364979c5beee91842f8f0ef42ab558e9d4?relay=wss%3A%2F%2Frelay.damus.io&secret=71a8c14c1407c113601079c4302dab36460f0ccd0ad506f1f2dc73b5100e4f3c
+```
+
+## Commands
+
+### `get_info`
+
+Description: Get information about the wallet and service.
+
+Request: Empty object.
+
+Response:
+```jsonc
+{
+    "balance": 100000, // balance in msat, int
+                       // this should be capped at the quota allowed for this client
+                       // to not report unspendable balance.
+    "implemented_commands": ["get_info", "pay_invoice"]  // commands supported, string array
+                            // extensions can be specified via command+extension:
+                            // get_info+node_info
+}
+```
+
+### `create_invoice`
+
+Description: Requests creation of an invoice.
+
+Request:
+```jsonc
+{
+    "amount": 1000, // amount in msat, int
```

Comment:
Use sats instead of msats.

Reply:
definitely should be in msats, no reason to lose granularity